doc/MiniLanguageTutorial.md

```diff
@@ -480,37 +563,278 @@ Mini never automatically converts types to make an operation succeed.  Programme
+* `nominal`, `specialized generic`:  
+  The default value for a nominal or specialized generic type is the default value of its representation.
+
+While non-specialized generic types should never be created, if they are they will have a default value of `()`.
+
+## AVM Representation
+
+Types in mini are internally represented as AVM values.
+
+1. uint:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. int:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. bool:  
+   Always an `Int`, the values `0` and `1` are allowed for this type, but nothing else.
+1. buffer:  
+   Always a `Buffer`, any value is allowable for this type.
+1. bytes32:  
+   Always an `Int`, any value is allowable for this type.
+1. address:  
+   Always an `Int`, only values up to `2^160-1` are allowed for this type, where the `Int` must be interpreted as unsigned.
+1. void:  
+   Represents lack of a value, whenever a value of type `void` is present, then it is assumed there is no value there.
+1. struct:  
+   Represented by a wrapping tuple containing values of the types of each of its fields, in order from top to bottom.
+   Field names have no impact on the allowable values for the type.
+1. tuple:  
+   Identical to structs, represented by a wrapping tuple containing values of the types of each of its fields, in order left to right.
+1. unsized array of *type*:  
+   The depth of an unsized array is the smallest integer `x` such that 8^`x` > *length*, with a minimum depth of `1`, where *length* is the current length of the array.
+   An array of depth `N` has a value of `Tuple(Int, Int,` a `Tuple` of length 8 of sized arrays of depth *N-1* `)`, where depth `0` represents an AVM value valid for *type*.
+   The first `Int` in the outer tuple must be the *length* of the array, and the second integer must be `8^(N-1)`.
+1. sized array of *type* and length *length*:  
+   A series of nested 8 tuples, each leaf of which contains a AVM value valid for *type*.
+   The nested tuples are at uniform depth, and are the minimum depth such that there are enough slots for *length* values.
+   The minimum depth is a single 8 tuple. Sized arrays of length 0 allocate a single tuple with 8 slots.
+1. map with keys of *keytype* and values of *valuetype*:  
+   All maps consist of `Tuple(` *KvsNode* `, Int)`, the valid values for *KvsNode* will be explained below.
+   A *KvsNode* may be one of 3 types, it may be an integer of value `0`, a `Tuple` of length 2, or a `Tuple` of length 8.
+   In the case of a 2 `Tuple`, the first value in the tuple must be valid for *keytype*, and the second must be valid for `option<`*valuetype*`>`.
+   In the case of an 8 tuple, each value in the tuple must be a valid *KvsNode*.
+1. string:  
+   As a string is equivalent to `(uint, buffer)`, the valid AVM values are those values that are valid for `(uint, buffer)`,
```

Comment:
Again, need to say how the values in the `(uint, buffer)` structure correspond to string values in the language.